prime-dotnet-webapi-tests/Unit Tests/RazorConverterService.Tests.cs

```diff
@@ -77,62 +81,150 @@ public async void TestRender_Agreement_WithLimits(RazorTemplate<Agreement> templ
+        {
+            var service = CreateService();
+            var model = new EnrolleeRenewalEmailViewModel("first", "last", DateTimeOffset.Now);
+
+            var html = await service.RenderTemplateToStringAsync(template, model);
+
+            Assert.NotNull(html);
+            Assert.Contains(model.EnrolleeName, html);
+            // Not all emails contain the renewal date or URL despite sharing a view mmodel.
+        }
+
+        [Theory]
+        [MemberData(nameof(SiteApprovalEmailTemplates))]
+        public async void TestRender_SiteApprovalEmails(RazorTemplate<SiteApprovalEmailViewModel> template)
+        {
+            var service = CreateService();
+            var model = new SiteApprovalEmailViewModel
+            {
+                DoingBusinessAs = "dba",
+                Pec = "pec"
             };
 
             var html = await service.RenderTemplateToStringAsync(template, model);
 
             Assert.NotNull(html);
+            Assert.Contains(model.DoingBusinessAs, html);
+            // Not all emails contain the PEC despite sharing a View Model.
+        }
+
+        [Fact]
+        public async void TestRender_RemoteUserNotificationEmail()
+        {
+            var service = CreateService();
+            var model = new RemoteUserNotificationEmailViewModel
+            {
+                OrganizationName = "NAme",
+                SiteStreetAddress = "2134 first street",
+                SiteCity = "Acity",
+                PrimeUrl = "www.TEST.com"
+            };
+
+            var html = await service.RenderTemplateToStringAsync(RazorTemplates.Emails.RemoteUserNotification, model);
+
+            Assert.NotNull(html);
+            Assert.Contains(model.OrganizationName, html);
+            Assert.Contains(model.SiteStreetAddress, html);
+            Assert.Contains(model.SiteCity, html);
+            Assert.Contains(model.PrimeUrl, html);
+        }
+
+        [Fact]
+        public async void TestRender_RemoteUsersUpdatedEmail()
+        {
+            var service = CreateService();
+            var model = new RemoteUsersUpdatedEmailViewModel
+            {
+                SiteStreetAddress = "123 street",
+                OrganizationName = "NAME",
+                SitePec = "PecC",
+                RemoteUserNames = new[] { "Alice", "bob bobward" },
+                DocumentUrl = "A.URL.com"
+            };
+
+            var html = await service.RenderTemplateToStringAsync(RazorTemplates.Emails.RemoteUsersUpdated, model);
+
+            Assert.NotNull(html);
+            Assert.Contains(model.SiteStreetAddress, html);
+            Assert.Contains(model.OrganizationName, html);
+            Assert.Contains(model.SitePec, html);
+            Assert.Contains(model.DocumentUrl, html);
+            foreach (var name in model.RemoteUserNames)
+            {
+                Assert.Contains(name, html);
+            }
         }
 
         [Fact]
         public async void TestRender_SiteSummary()
         {
```

Comment:
Great that you've added all these unit tests but can `TestRender_SiteSummary` verify more of the generated `SiteRegistrationReview` content?

Reply:
Most (all?) of the other razor templates have small ViewModels and use the data semantically; so we expect to see it in the rendered html with little variation.  The site registration review, however, is more of a collection of almost every piece of data about a site, and so is likely to change if we add or remove anything to the site Model. I was hesitant to test that any specific data was included in the rendered html lest the test becomes overly brittle.

Reply:
It would just be nice to verify that some semblance of the expected document is produced

Reply:
@james-hollinger As we talked about on Teams, I've updated `TestRender_SiteSummary`.  I also updated `prime-dotnet-webapi-tests\README.md` after testing CHES client